### PR TITLE
docs(book): remote ExEx chapter

### PR DESCRIPTION
**Pull request by shekhirin**
Ref https://github.com/paradigmxyz/reth/pull/8890

### CODE REVIEW

File: book/developers/exex/remote.md [hunk 1]
```diff
@@ -1,3 +1,489 @@
...
+[[bin]]
+name = "exex"
+path = "src/exex.rs"
+
+[[bin]]
+name = "consumer"
+path = "src/consumer.rs"
+```
+
+As you can see, we also added a build dependency for Tonic. We will use it to generate the Rust code for our
```
Comment (mattsse) [inline book/developers/exex/remote.md:65]:
```suggestion
We also added a build dependency for Tonic. We will use it to generate the Rust code for our
```

File: book/developers/exex/remote.md [hunk 2]
```diff
@@ -1,3 +1,489 @@
...
+Protobuf definitions at compile time.
+
```
Comment (mattsse) [inline book/developers/exex/remote.md:67]: can we link to the tonic docs here as well?

**Review by mattsse — CHANGES_REQUESTED** [review]
this is pretty cool, I worry about keeping this up to date because these are just docs and will most likely break sooner than later.

is there a way we could include! this file in a crate and enforce it compiles

File: book/developers/exex/remote.md (hunk 2 above)
Comment (shekhirin) [reply book/developers/exex/remote.md:68]: Added a link to their hello world tutorial

**Comment by Rjected** [comment]
Found this in mdBook for including files, maybe we could use it:
https://rust-lang.github.io/mdBook/format/mdbook.html#including-files

**Comment by shekhirin** [comment]
@Rjected yeah nice, I think we can have several intermediate `.rs` files and a final one that we'll compile as part of examples, and then include them in the book using `{{#include file.rs}}`

**Comment by shekhirin** [comment]
Hmm, we moved this example to a separate repo to get rid of a dependency on `protoc`, so there's no way to do the `include` now I'm afraid. Unless we want to clone this repo into the book dir in CI and include the code from there.

**Comment by mattsse** [comment]
makes sense

**Review by mattsse — APPROVED** [review]

**Pull request merged**